pkg/network/server.go

```diff
@@ -1042,19 +1049,39 @@ func (s *Server) handleTxCmd(tx *transaction.Transaction) error {
 	}
 	s.txInMap[tx.Hash()] = struct{}{}
 	s.txInLock.Unlock()
-	s.serviceLock.RLock()
-	txCallback := s.txCallback
-	s.serviceLock.RUnlock()
-	if txCallback != nil && s.txCbEnabled.Load() {
-		txCallback(tx)
+	s.txin <- tx
+	return nil
+}
+
+func (s *Server) txHandlerLoop() {
+txloop:
+	for {
+		select {
+		case tx := <-s.txin:
+			s.serviceLock.RLock()
+			txCallback := s.txCallback
+			s.serviceLock.RUnlock()
+			if txCallback != nil && s.txCbEnabled.Load() {
+				txCallback(tx)
+			}
+			if s.verifyAndPoolTX(tx) == nil {
+				s.broadcastTX(tx, nil)
+			}
+			s.txInLock.Lock()
+			delete(s.txInMap, tx.Hash())
+			s.txInLock.Unlock()
+		case <-s.quit:
+			break txloop
+		}
 	}
-	if s.verifyAndPoolTX(tx) == nil {
-		s.broadcastTX(tx, nil)
+drainloop:
+	for {
+		select {
+		case <-s.txin:
+		default:
+			break drainloop
```

Comment:
And close `txin` after that?

Reply:
`txHandlerLoop()` can't do that, we have a number of them. We don't close `s.transactions` also, GC will handle it, although we only do this when server shuts down and application exits.